streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -61,7 +61,7 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
             List<PartitionInfo> infos = metadata.partitionsForTopic(topic);
 
             if (infos == null)
-                throw new KafkaException("topic not found :" + topic);
+                throw new KafkaException("Topic not found during partition assignment: " + topic);
```

Comment:
Again, why not a specific type? More generally, should we ever have a bare `KafkaException` in this code? It'd be nice to set expectations up front, e.g. all Kafka Streams code will either throw one of the normal built-ins (like IllegalStateException) or a Kafka Streams-specific exception. Not sure that's actually the policy we want, but right now I don't understand the policy since we're still mixing Kafka Streams exceptions and regular KafkaExceptions.